test/mpi/maint/gen_all_tests.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+set -xe
+srcdir=$(dirname $0)
+
+dtp_types="MPI_INT,MPI_INT:4+MPI_DOUBLE:8"
+$srcdir/gentests_dtp.sh --with-dtpools-datatypes=$dtp_types
+
+find . -name testlist.gpu | xargs rm -f
+for a in host reg_host device ; do
+    for b in host reg_host device ; do
+        if test $a = device -o $b = device ; then
+            $srcdir/gentests_dtp.sh --with-dtpools-datatypes=$dtp_types --with-dtpools-memtype=$a,$b,random --append
```

Comment:
Does this mean we no longer have testlist.gpu ?
Currently we still need this file because we would like to separate GPU tests from the rest of the tests so that for a RPM build with GPU enabled but running on a host without GPU device, we can disable all GPU tests in the testlist.gpu easily.

Reply:
We will have testlist.gpu. Above code just says we skip "[reg_]host" to "[reg_]host" in `testlist.gpu`.